migration/migrate_db.go

```diff
@@ -0,0 +1,93 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/go-pg/pg/v10"
+	migrate "github.com/golang-migrate/migrate/v4"
+	_ "github.com/golang-migrate/migrate/v4/database/postgres"
+	_ "github.com/golang-migrate/migrate/v4/source/file"
+)
+
+func isEnvExist(key string) bool {
+	if _, ok := os.LookupEnv(key); ok {
+		return true
+	}
+
+	return false
+}
+
+func checkConn(db *pg.DB) error {
+	var n int
+	_, err := db.QueryOne(pg.Scan(&n), "SELECT 1")
+	return err
+}
+
+func getAddrandPassword() (string, string, int) {
+	port := 5432
+	addr := "localhost"
+	if isEnvExist("DB_ADDR") {
+		addr = os.Getenv("DB_ADDR")
+	}
+
+	password := "gitops"
+	if isEnvExist("DB_PASS") {
+		password = os.Getenv("DB_PASS")
+	}
+	return addr, password, port
+}
+
+func connectToDatabaseWithPort() (*pg.DB, error) {
+	addr, password, port := getAddrandPassword()
+	opts := &pg.Options{
+		Addr:     fmt.Sprintf("%s:%v", addr, port),
+		User:     "postgres",
+		Password: password,
+		Database: "postgres",
+	}
+
+	db := pg.Connect(opts)
+
+	if err := checkConn(db); err != nil {
+		return nil, fmt.Errorf("%v, unable to connect to database: Host:'%s' User:'%s' Pass:'%s' DB:'%s' ", err, opts.Addr, opts.User, opts.Password, opts.Database)
+	}
+
+	return db, nil
+}
+
+func main() {
```

Comment:
yeah, so point of having package main is for it to run separately as it is basically meant to be used as a cli tool instead of trying to import it anywhere as it updates the schema of the db.

Reply:
to my understanding talking to @tirthyme, the intention here is to have this as a standalone binary/executable. This code is not meant to be exported and consumed by other packages. The use case is to do something like this from the Makefile:

```shell
cd hacks/migration; go run migrate_db.go
```

To do that, the package has to be `main`. It's a common pattern to create isolated files for your code:

```shell
/ (root)
 -> examples (dir)
      -> ex1 (dir)
            ex1.go (main pkg)
      -> ex2 (dir)
             ex2.go (main pkg)
      -> ex3 (dir)
             ex3.go (main pkg)
```

Ideally, we can introduce a `cmd` package for the CLI stuff. In that case, yes it could name the package `db` because it would have been imported by the `cmd` one. Right now we have just the make file where is the common denominator for calling all of these _hacks_.

Reply:
These are quite generic db util functions which probably will get reused as some point. I think we should do either the hack pattern or make it like a CLI as @drpaneas described. Either way is not hard. :-)